app/webapp/src/main/java/uk/ac/ebi/fg/annotare2/web/gwt/editor/client/view/experiment/design/DataAssignmentViewImpl.java

```diff
@@ -176,28 +177,36 @@ private String getDataFileColumnName(FileType type, int index) {
 
     private List<EnumWithHelpText> getAllowedColumnTypes() {
         List<EnumWithHelpText> types = new ArrayList<>();
-        for (FileType type : FileType.values()) {
-            if(!type.isProcessedMatrix()){
-                if (experimentType.isSequencing()){
-                    types.add(type);
-                }
-                else if (isMicroArrayOrTwoColourMicroArrayWithNonRawMatrixFileType(type)){
-                    alwaysAddProcessedTypeToListAndOnlyAddOtherTypesOnce(type, types);
-                }
-            }
+        if(experimentType.isSequencing()){
```

Comment:
These if statements can be easily replace by applying design patterns.
I let you decide which one is better:
- Command Pattern
- Strategy + Factory Method
- Enums

Thanks